src/security/ServerOptions.cc

```diff
@@ -372,9 +376,36 @@ Security::ServerOptions::loadDhParams()
             dhp = nullptr;
         }
     }
-
     parsedDhParams.resetWithoutLocking(dhp);
+
+#else // OpenSSL 3.0+
+    EVP_PKEY *pkey = nullptr;
+    const char *type = "DH";
+    if (!eecdhCurve.isEmpty())
+        type = "EC";
+    // XXX: use the eecdhCurve name when generating the EVP_KEY object. or at least verify it matches the loaded params.
+
+    if (auto *dctx = OSSL_DECODER_CTX_new_for_pkey(&pkey, "PEM", nullptr, type, OSSL_KEYMGMT_SELECT_ALL, nullptr, nullptr)) {
```

Comment:
Is that XXX a regression (i.e. we were properly handing eecdhCurve before these changes) or just a missing feature (i.e. we did not use eecdhCurve name before these changes either)? This change request does not request any code changes. I just do not know enough about this code to guess the answer by reading the XXX comment itself.

Reply:
This is a regression that needs to be resolved before this PR is suitable for use by anyone wanting EC ciphers.

Reply:
It turns out the issue was OSSL_KEYMGMT_SELECT_ALL causing errors when irrelevant (to EC) details were missing from the parameters file. Patch fixing that and polishing debugs output coming soon as the test-builds.sh completes here.

OpenSSL auto-detects the curve name etc from the file contents. It is not clear why earlier versions needed the curve name to be provided by the load functions (validation maybe?). I'm reducing this comment to a TODO for later.